beamer/events.py

```diff
@@ -285,7 +285,8 @@ def _fetch_range(
 
     def fetch(self) -> list[Event]:
         try:
-            block_number = BlockNumber(self._web3.eth.block_number)
+            block_data = self._web3.eth.get_block("latest")
+            block_number = BlockNumber(block_data["number"])
```

Comment:
This is nice. If we take care to just return one `LatestBlockUpdatedEvent` from `fetch`, we should be able to reuse the data we got here during construction of the `LatestBlockUpdatedEvent` at the end of this function. We can thereby save another `get_block` call and unnecessary work.

Reply:
see my comment above